lib/utils.js

```diff
@@ -689,22 +689,27 @@ function* makeCounter(seed) {
 /**
  * Helper function for either accepting a callback, or returning a promise
  *
- * @param {Function} [callback] an optional callback.
- * @param {Function} fn A function that takes a callback
+ * @param {Object} parent an instance of parent with promiseLibrary.
+ * @param {object} parent.s an object containing promiseLibrary.
+ * @param {function} parent.s.promiseLibrary an object containing promiseLibrary.
+ * @param {function | undefined} callback The last function argument in exposed method, controls if a Promise is returned
```

Comment:
I'm using union `undefined` here because it's not optional.

Reply:
:+1:

Reply:
I think we can use `{?function}` here to indicate it is [nullable type](https://github.com/jsdoc/jsdoc/blob/master/packages/jsdoc/lib/jsdoc/tag/type.js#L122-L123).